wandb/sdk/wandb_manager.py

```diff
@@ -1,235 +0,0 @@
-        assert version == self._version
-        assert transport in self._supported_transports
-        self._pid = int(pid_str)
-        self._transport = transport
-        self._host = host
-        self._port = int(port_str)
-
-    def reset_environment(self) -> None:
-        os.environ.pop(env.SERVICE, None)
-
-    @property
-    def token(self) -> str:
-        return self._token_str
-
-    @property
-    def pid(self) -> int:
-        return self._pid
-
-    @property
-    def transport(self) -> str:
-        return self._transport
-
-    @property
-    def host(self) -> str:
-        return self._host
-
-    @property
-    def port(self) -> int:
-        return self._port
-
-
-class _Manager:
-    _token: _ManagerToken
-    _atexit_lambda: Optional[Callable[[], None]]
-    _hooks: Optional[ExitHooks]
-    _settings: "Settings"
-    _service: "service._Service"
-
-    def _service_connect(self) -> None:
-        port = self._token.port
-        svc_iface = self._get_service_interface()
-
-        try:
-            svc_iface._svc_connect(port=port)
-
-        except ConnectionRefusedError as e:
-            if not psutil.pid_exists(self._token.pid):
-                message = (
-                    "Connection to wandb service failed"
-                    " because the process is not available."
-                )
-            else:
-                message = "Connection to wandb service failed."
-            raise ManagerConnectionRefusedError(message) from e
-
-        except Exception as e:
-            raise ManagerConnectionError(
-                "Connection to wandb service failed.",
-            ) from e
-
-    def __init__(self, settings: "Settings") -> None:
-        """Connects to the internal service, starting it if necessary."""
-        from wandb.sdk.service import service
-
-        self._settings = settings
-        self._atexit_lambda = None
-        self._hooks = None
-
-        self._service = service._Service(settings=self._settings)
-
-        token = _ManagerToken.from_environment()
-        if not token:
-            self._service.start()
-            host = "localhost"
-            transport = "tcp"
-            port = self._service.sock_port
-            assert port
-            token = _ManagerToken.from_params(transport=transport, host=host, port=port)
-            token.set_environment()
-            self._atexit_setup()
-        self._token = token
-
-        try:
-            self._service_connect()
-        except ManagerConnectionError as e:
-            wandb._sentry.reraise(e)
-
-    def _teardown(self, exit_code: int) -> int:
-        """Shuts down the internal process and returns its exit code.
-
-        This sends a teardown record to the process. An exception is raised if
-        the process has already been shut down.
-        """
-        unregister_all_post_import_hooks()
-
-        if self._atexit_lambda:
-            atexit.unregister(self._atexit_lambda)
```

Comment:
also i didn't see where this code is now

Reply:
This was missing; added it back in `service_connection.py::_start_and_connect_service`.